domain/application/errors/errors.go

```diff
@@ -58,6 +58,10 @@ const (
 	// operated on does not have a status.
 	UnitStatusNotFound = errors.ConstError("unit status not found")
 
+	// UnitProviderIDNotFound describes an error that occurs when the unit provider
+	// id is not found.
+	UnitProviderIDNotFound = errors.ConstError("unit provider id not found")
```

Comment:
Seems weird, and it backs up my previous comment: maybe we should have a method that just returns the full pod, in that case the error would be "pod not found".

Reply:
Removed